### ISSUE TITLE
Enabling new messages for Discussions

### ISSUE THREAD

**Issue by stefanos82** [transmission/transmission#1807]
Can we make Discussions accessible to users?

I wanted to open a discussion thread about @kvakvs 's PR #1801 and expand it even further.

There are lots of interesting ideas that would need to be discussed first and then get approved or rejected by Transmission's core developers.

**Comment by kvakvs** [comment]
I will emphasize how important is for core devs to agree on #1801 which then I can follow, to minimize the style discussions and nit picking in each partial PR. Also not moving ahead till that one is cleared.

**Comment by stefanos82** [comment]
Of course, I completely agree.

The reason I want the Discussions open to the public is so we can start discussing possible ways to set some TODO tasks assigned to contributors or volunteers.

For instance, we could break down each TODO task to micro-tasks and make it easier to resolve issues or refactor legacy code to modern C++, bit by bit without affecting the review process by core developers.

To give an example, current task was to refactor `MIN()` to `std::min()`.

When I started with a local fork of mine, I did so many changes at once but didn't push, because I read the feedback from core developers they were exchanging between them and agreed with them for wanting to review as less lines of code as possible so they can make their lives easier; I understand that.

Thus my suggestion(s) below.

We have to have the necessary tasks in advance so we can evaluate the time it would take to deliver *and* agree on the methods we are going to adopt as a community; something like a developer's guidelines in a way.

An example would be, contributor `A` will be dealing with replacing `qsort` with `std::sort`, contributor `B` will be dealing with refactoring `malloc()` / `free()` to `new` / `delete`, and when both contributors finish, they can start discussing how to move code to modern methodologies to deal with objects, such as `unique_ptr` / `shared_ptr`...you get the idea.

What do you think?

**Comment by kvakvs** [comment]
The suggested modernization stages in `#1801` are composable and can be done on separate files by different people. One person can do one or a few stages, merge, and then a new PR can be made. And this can be parallelized for multiple files.

There is cost of review and merge, from what i see now it takes weeks per PR. Making this to be hundreds of small PRs is going to have exponential review cost lasting years. I would prefer to do full cycle on one file (first would be `announcer.cc` as it is first in alphabet, and it would be moved into a subdirectory, split, and all functions will churn and change ownerships, and it will probably be a 2500-3000 line review (because announcer is a 2000 line file), but at least it needs to be done only once per file.

**Comment by stefanos82** [comment]
OK, let's wait to hear what core developers have to say.

**Comment by ckerr** [comment]
I've enabled discussions for now to see how it goes. My main concerns are:

- There is already an existing forum at https://forum.transmissionbt.com/ and I don't know whether or not discussions will make things more confusing about where discussion should take place.
- I don't know how much work this will involve for maintainers. As people who've watched the project know, this is a very small team right now and one wonderful thing about forum.transmissionbt.com is that it's policed by other users rather than maintainers.
- The project has had some problems with toxic users in the past and I don't know what kind of moderation tools are available.

That said... I don't think either of you are on that forum, and it's not particularly intuitive to aim people for some discussion about development work. I'm willing to see how this goes for a little while.

**Comment by stefanos82** [comment]
To be honest with you, I didn't even know there was a forum and I had to go to website to find it under the `Support`.

If Discussions will affect the existing community or you think it could divide it, please by all means then close it and just leave a message like a sticky post or something to point them at forum URL.
